presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -121,6 +121,9 @@ public boolean add(Block block, int position)
 
         // containsNullElement flag is maintained so contains() method can have shortcut for null value
         if (block.isNull(position)) {
+            if (containNullElements) {
+                return false;
```

Comment:
Are you sure we don't use it for things that can have nulls?

Reply:
This should handle nulls correctly. Previously when null was added again, it would redo the processing and ignore it. 

Now when nulls are already added (i.e. containNullElements) and re adding null, it short circuits and returns false.